component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -144,120 +158,65 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                 throw new AuthenticationFailedException
                         ("Authentication failed!. Cannot proceed further without identifying the user");
             }
+            boolean isUserExistence = federatedAuthenticator.isExistUserInUserStore(username);
+            boolean isEmailOTPEnabledByUser = EmailOTPUtils.isEmailOTPEnableForLocalUser(username, context, isUserExistence);
             String queryParams = FrameworkUtils.getQueryStringWithFrameworkContextId(
                     context.getQueryParams(), context.getCallerSessionKey(),
                     context.getContextIdentifier());
             String retryParam = "";
             String errorPage = emailOTPParameters.get(EmailOTPAuthenticatorConstants.EMAILOTP_AUTHENTICATION_ERROR_PAGE_URL);
-            boolean isEmailOTPMandatory = Boolean.parseBoolean(emailOTPParameters
-                    .get(EmailOTPAuthenticatorConstants.IS_EMAILOTP_MANDATORY));
-            boolean isEmailOTPEnabledByUser = EmailOTPUtils.isEmailOTPEnableForLocalUser(username, context);
-            // Email OTP authentication is mandatory and user doesn't have Email value in user's profile.
-            if (isEmailOTPMandatory && !isEmailOTPEnabledByUser) {
-                // that Enable the EMAIL OTP in user's Profile. Cannot proceed further without EmailOTP authentication.
-                retryParam = EmailOTPAuthenticatorConstants.ERROR_EMAILOTP_DISABLE;
-                if (StringUtils.isEmpty(errorPage)) {
-                    errorPage = ConfigurationFacade.getInstance().getAuthenticationEndpointURL()
-                            .replace(EmailOTPAuthenticatorConstants.LOGIN_PAGE, EmailOTPAuthenticatorConstants.ERROR_PAGE);
-                    if (log.isDebugEnabled()) {
-                        log.debug("Default authentication endpoint context is used");
+            if (isEmailOTPMandatory) {
```

Comment:
This is not readable. Please avoid nested if / else blocks and try to resolve it elseif etc.